pkg/limayaml/validate.go

```diff
@@ -109,6 +109,10 @@ func Validate(y LimaYAML, warn bool) error {
 				i, ProvisionModeSystem, ProvisionModeUser)
 		}
 	}
+	needsContainerdArchives := (y.Containerd.User != nil && *y.Containerd.User) || (y.Containerd.System != nil && *y.Containerd.System)
```

Comment:
If we support calling `Validate()` without calling `FillDefaults()` first (and I think it is good to be robust here to avoid a panic, even though virtually all `lima.yaml` files will fail validation without `FillDefaults`), then `y.UseHostResolver` should be checked for `nil` as well before it is dereferenced.

Reply:
fixed